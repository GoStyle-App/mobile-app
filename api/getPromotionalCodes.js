export default function getPromotionalCodes() {
    const url = 'http://goreduc.herokuapp.com/api/promotional_codes';
    return fetch(url)
        .then((response) => response.json())
        .then((json) => json['hydra:member'])
        .catch((error) => console.error(error));
}
