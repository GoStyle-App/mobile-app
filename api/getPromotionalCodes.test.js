require('jest-fetch-mock').enableMocks();
import React from 'react';
import getPromotionalCodes from './getPromotionalCodes';

describe('Getting promotionnal codes without problem.', () => {
    fetch.mockResponse(JSON.stringify({
        '@context': '/api/contexts/PromotionalCode',
        '@id': '/api/promotional_codes',
        '@type': 'hydra:Collection',
        'hydra:member': [
            {
                '@id': '/api/promotional_codes/1',
                '@type': 'PromotionalCode',
                'id': 1,
                'label': 'Promotion été 2020',
                'description': 'Bénéficier de 20% sur vos achats cet été',
                'code': 'ETE2020',
                'startDate': '2021-04-24T00:00:00+00:00',
                'endDate': '2021-05-30T00:00:00+00:00'
            },
            {
                '@id': '/api/promotional_codes/2',
                '@type': 'PromotionalCode',
                'id': 2,
                'label': 'Promotion décembre',
                'description': 'Bénéficiez de 15% sur votre panier, pour tout achat supérieur à 50€. Cette offre est valable sur tous les articles de notre boutique GoStyle.',
                'code': '15SKATES',
                'startDate': '2021-02-04T00:00:00+00:00',
                'endDate': '2021-06-10T00:00:00+00:00'
            },
            {
                '@id': '/api/promotional_codes/5',
                '@type': 'PromotionalCode',
                'id': 5,
                'label': 'Promos chaussures',
                'description': 'Profitez d\'une méga promo de 10% sur tous vos articles préférés',
                'code': 'CHAUSS5',
                'startDate': '2021-03-01T00:00:00+00:00',
                'endDate': '2021-03-31T00:00:00+00:00'
            }
        ],
        'hydra:totalItems': 4
    }));

    test('API call return data', async () => {
        const response = await getPromotionalCodes();

        expect(response).not.toBeNull();
    });

    test('API call return all PromotionalCodes', async () => {
        const response = await getPromotionalCodes();
        const promotionalCodes = [];

        response.forEach((promotionalCode) => {
            promotionalCodes.push(promotionalCode);
        });

        expect(promotionalCodes.length).toEqual(3);
    });
});
