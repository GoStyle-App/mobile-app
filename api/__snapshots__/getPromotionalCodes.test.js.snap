// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting promotionnal codes without problem. CORRECTLY RENDER LIST OF PROMOTIONNAL CODES 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 80,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2E2E2E",
        "fontSize": 20,
        "marginBottom": 30,
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    Liste des codes promotionnels scanables
  </Text>
  <Text>
    Pas de codes promos
  </Text>
</View>
`;
